core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/TupleExprBuilder.java

```diff
@@ -310,7 +316,7 @@ protected Var createAnonVar() {
 		// the
 		// varname
 		// remains compatible with the SPARQL grammar. See SES-2310.
-		final Var var = new Var("_anon_" + UUID.randomUUID().toString().replace("-", "_"));
+		final Var var = new Var("_anon_" + uniqueIdPrefix + uniqueIdSuffix.incrementAndGet());
```

Comment:
I was first thinking this would get you in trouble because you're no longer replacing the '-'. But then spotted you do it right at the start. Clever.